aioredis/commands/cluster.py

```diff
@@ -7,94 +9,281 @@ class ClusterCommandsMixin:
+        return wait_ok(fut)
+
+
+def _decode(s, encoding):
+    if encoding:
+        return decode(s, encoding)
+    return s
+
+
+def parse_info(resp, **options):
+    e = options.get('encoding')
+
+    lines = resp.strip().splitlines()
+    result = list(map(lambda s: s.split(_decode(b':', e)), lines))
+    return dict(result)
+
+
+def parse_node_slots(raw_slots, encoding=None):
+    """
+    @see: https://redis.io/commands/cluster-nodes#serialization-format
+    @see: https://redis.io/commands/cluster-nodes#special-slot-entries
+    """
+
+    slots, migrations = [], []
+    r_delimiter = _decode(b'->-', encoding)
+    l_delimiter = _decode(b'-<-', encoding)
+    m_delimiter = _decode(b'-', encoding)
+    migrating_state = _decode(b'migrating', encoding)
+    importing_state = _decode(b'importing', encoding)
+
+    for r in raw_slots.strip().split():
+        if r_delimiter in r:
+            slot_id, dst_node_id = r[1:-1].split(r_delimiter, 1)
+            migrations.append({
+                'slot': int(slot_id),
+                'node_id': dst_node_id,
+                'state': migrating_state
+            })
+        elif l_delimiter in r:
+            slot_id, src_node_id = r[1:-1].split(l_delimiter, 1)
+            migrations.append({
+                'slot': int(slot_id),
+                'node_id': src_node_id,
+                'state': importing_state
+            })
+        elif m_delimiter in r:
+            start, end = r.split(m_delimiter)
+            slots.append((int(start), int(end)))
+        else:
+            slots.append((int(r), int(r)))
+
+    return tuple(slots), tuple(migrations)
+
+
+def parse_cluster_nodes_lines(lines, encoding=None):
+    """
+    @see: http://redis.io/commands/cluster-slaves # list of string
+    """
+
+    address_splitter = _decode(b':', encoding)
+    port_splitter = _decode(b'@', encoding)
+    flags_splitter = _decode(b',', encoding)
+    m_delimiter = _decode(b'-', encoding)
+
+    for line in lines:
+        parts = line.split(None, 8)
+        self_id, addr, flags, master_id, ping_sent, \
+            pong_recv, config_epoch, link_state = parts[:8]
+
+        host, port = addr.rsplit(address_splitter, 1)
+        nat_port = None
+
+        if port_splitter in port:
+            # Since version 4.0.0 address_node_info has the format
+            # '192.1.2.3:7001@17001
+            at_index = port.index(port_splitter)
+            nat_port = int(port[at_index + 1:])
+            port = port[:at_index]
+
+        node = {
+            'id': self_id,
+            'host': host,
+            'port': int(port),
+            'nat-port': nat_port,
+            'flags': tuple(flags.split(flags_splitter)),
+            'master': master_id if master_id != m_delimiter else None,
+            'ping-sent': int(ping_sent),
+            'pong-recv': int(pong_recv),
+            'status': link_state,
+            'slots': tuple(),
+            'migrations': tuple(),
+        }
```

Comment:
Is there are reason why config_epoch is not returned?

Reply:
I miss it (